### DISCUSSION TITLE
External library vs. upload via CLI

### DISCUSSION THREAD

**Discussion by kaibagley** [immich-app/immich#10925] [Q&A] [answered]
Hey guys,

I might be misunderstanding something here but it seems to me that Immich will scan an external library, and basically copy it to the Immich "internal" library and keep it's internal library in sync with the external. However when i upload using the CLI it will add it to the internal library and that's that.

If I have a directory filled with photos, and I want to have them all loaded into Immich without worrying about anything external to Immich editing it, what is the best way to do this? External library, CLI upload or something else?

Thanks!

**Comment by bo0tzz** [comment, marked as answer]
> copy it to the Immich "internal" library

No, if you use an external library then the files stay where they are.

The only reason to use an external library is if you need to manage the files from outside of Immich, eg from another program. If you want Immich to manage your files, then upload them with the CLI.

**Reply by kaibagley** [reply]
Ah I see, thank you.

Are there any important differences between the immich-cli and immich-go? Going by the GH page for immich-go it looks like it detects, dates, bursts etc. Does the CLI do this too? I can't see too many details on features for the CLI

**Reply by bo0tzz** [reply]
The CLI is very simple - it just uploads files. Immich-go is (primarily) built for parsing and uploading Google Takeout files, but these days I believe it also does 'simple' file uploads.

**Reply by kaibagley** [reply]
Thanks so much, I ended up going with immich-go because I was having problems with npm, and its working beautifully.
